proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,190 @@
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
+1. default value?
+
+   Should we add nuget.org as a default value, so that customers who remove nuget.org as a package source are more easily able to use NuGetAudit?
+   Would using nuget.org as a default audit source cause undue concern from customers who don't want to use nuget.org as a package source, and the new tooling version starts making network requests to nuget.org?
+
+   If we want to make nuget.org a default/automatic audit source, how?
+   NuGet has tried before to use a tracking file to automatically add nuget's v3 URL as a package source to user-profile nuget.config files that don't have any package sources.
+   However, this caused a security issue when customers wanted to remove nuget.org as a default package source, and this code incorrectly re-added it once (the customer could successfully remove nuget.org a second time).
+   When we removed this auto-update to mitigate the security concern, it exposed a bug where multiple other apps created user-profile nuget.config files without any package sources, and many customers assumed it was a bug with NuGet.
+
+   Should we hard-code nuget.org's default URL in NuGet.Configuration, and require customers to use `<clear/>` if they don't want any audit source?
+
+   Should we just add it as a default in new user-profile nuget.config files, and just accept that existing developers won't get it by default?
+
+2. Stop using package source as audit source?
+
+   On one hand, using package sources as a defacto audit source makes it easier for customers to use the feature when they knowingly or unknowingly use a package source with a VDB.
+   On the other hand, overloading `<packageSources>` with multiple meanings is confusing and customers probably won't understand why sometimes they get vulnerable package information and sometimes they don't.
+   If we add nuget.org as a default audit source, and don't use packageSources when auditSources has a value, then using packageSources to get a VDB when auditSources is empty will be particularly confusing/unintuitive to customers.
+
+   Another challenge is that if we want to change the behavior it' will be a breaking change, which is typically discouraged.
+   However, given that nuget.org is the only know source with a VDB, if we make nuget.org a default audit source, then it will effectively be a non-breaking change.
+
+3. Tooling
+
+   How important is it to add CLI commands to manage audit sources?
+   How important is it to add a GUI in Visual Studio to manage audit sources?
```

Comment:
Another spec could cover this in a future iteration, IMO. Certainly, some indicator whether any audit source is available might be interesting. Does a UI showing 0 vulnerabilities mean: 
"there's no vulnerabilities or there's no audit source active in my view?"

Reply:
This spec is intended to be specific about being able to audit sources which are not package sources. Other generic NuGetAudit features were intentionally out of scope.

Your suggestion overlaps greatly with something already proposed in the original NuGetAudit (whole of feature) spec: https://github.com/NuGet/Home/blob/dev/proposed/2022/vulnerabilities-in-restore.md#command-line-restore-output

We also have a (private) tracking issue: https://github.com/NuGet/Client.Engineering/issues/2340  Why is it a private tracking issue? I don't know. Seems like an oversight now. Customers should be able to upvote the idea to give us feedback they think it's more or less important than other features.